test/CMakeLists.txt

```diff
@@ -5,27 +5,27 @@ find_package(catkin REQUIRED COMPONENTS class_loader)
 
 include_directories(${catkin_INCLUDE_DIRS})
 
-add_library(${PROJECT_NAME}_TestPlugins1 EXCLUDE_FROM_ALL plugins1.cpp)
+add_library(${PROJECT_NAME}_TestPlugins1 SHARED EXCLUDE_FROM_ALL plugins1.cpp)
 target_link_libraries(${PROJECT_NAME}_TestPlugins1 ${PROJECT_NAME})
 class_loader_hide_library_symbols(${PROJECT_NAME}_TestPlugins1)
-add_library(${PROJECT_NAME}_TestPlugins2 EXCLUDE_FROM_ALL plugins2.cpp)
+add_library(${PROJECT_NAME}_TestPlugins2 SHARED EXCLUDE_FROM_ALL plugins2.cpp)
 target_link_libraries(${PROJECT_NAME}_TestPlugins2 ${PROJECT_NAME})
 class_loader_hide_library_symbols(${PROJECT_NAME}_TestPlugins2)
 
 catkin_add_gtest(${PROJECT_NAME}_utest utest.cpp)
 if(TARGET ${PROJECT_NAME}_utest)
-  target_link_libraries(${PROJECT_NAME}_utest ${Boost_LIBRARIES} ${class_loader_LIBRARIES})
```

Comment:
It is possible that I've removed ${Boost_LIBRARIES} for some reason, but perhaps for the sake of minimizing the diff, can we remove these following changes? Unless it breaks a build of course.

Reply:
I think I actually removed them. You are right, we should keep the changes minimal. I reverted the change and rebased to keep the history clean.